test/extended/builds/s2i_extended_build.go

```diff
@@ -16,57 +16,65 @@ var _ = g.Describe("[builds][Slow] s2i extended build", func() {
 	defer g.GinkgoRecover()
 
 	var (
-		oc                 = exutil.NewCLI("extended-build", exutil.KubeConfigPath())
-		testDataDir        = exutil.FixturePath("testdata", "build-extended")
-		scriptsFromRepoBc  = filepath.Join(testDataDir, "bc-scripts-in-repo.json")
-		scriptsFromUrlBc   = filepath.Join(testDataDir, "bc-scripts-by-url.json")
-		scriptsFromImageBc = filepath.Join(testDataDir, "bc-scripts-in-the-image.json")
+		oc                    = exutil.NewCLI("extended-build", exutil.KubeConfigPath())
+		testDataDir           = exutil.FixturePath("testdata", "build-extended")
+		runnerConf            = filepath.Join(testDataDir, "jvm-runner.yaml")
+		runnerWithScriptsConf = filepath.Join(testDataDir, "jvm-runner-with-scripts.yaml")
+		scriptsFromRepoBc     = filepath.Join(testDataDir, "bc-scripts-in-repo.yaml")
+		scriptsFromUrlBc      = filepath.Join(testDataDir, "bc-scripts-by-url.yaml")
+		scriptsFromImageBc    = filepath.Join(testDataDir, "bc-scripts-in-the-image.yaml")
 	)
 
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// we have to wait until image stream tag will be available, otherwise
+		// `oc start-build` will fail with 'imagestreamtags "wildfly:10.0" not found' error.
+		// See this issue for details: https://github.com/openshift/origin/issues/10103
+		g.By("waiting when image stream tag for wildfly will be available")
+		wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
sorry i mixed up the methods (which is my fault because i named waitforopenshiftnamespaceimagestreams poorly, feel free to rename it in this PR, something like "WaitForOpenshiftNamespaceImageStreamsAndTags" would be better).

but i still stand by my comment that we need to sort out why WaitForAnImageStreamTag didn't work for you, because if it's not working and it's being used elsewhere, that's going to be a source of flakes.